diff --git a/api/BUILD.bazel b/api/BUILD.bazel
index 4c5e750..515f823 100644
--- a/api/BUILD.bazel
+++ b/api/BUILD.bazel
@@ -9,9 +9,9 @@ java_library(
     ]),
     visibility = ["//visibility:public"],
     deps = [
-        artifact("com.google.code.findbugs:jsr305"),
-        artifact("com.google.errorprone:error_prone_annotations"),
-        artifact("com.google.guava:failureaccess"),  # future transitive dep of Guava. See #5214
-        artifact("com.google.guava:guava"),
+        artifact("com.google.code.findbugs:jsr305", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.errorprone:error_prone_annotations", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.guava:failureaccess", repository_name = "contrib_rules_jvm_deps"),  # future transitive dep of Guava. See #5214
+        artifact("com.google.guava:guava", repository_name = "contrib_rules_jvm_deps"),
     ],
 )
diff --git a/compiler/BUILD.bazel b/compiler/BUILD.bazel
index ed8e946..8148df7 100644
--- a/compiler/BUILD.bazel
+++ b/compiler/BUILD.bazel
@@ -24,9 +24,9 @@ java_library(
         "//protobuf",
         "//stub",
         "//stub:javax_annotation",
-        artifact("com.google.code.findbugs:jsr305"),
-        artifact("com.google.guava:guava"),
-        "@com_google_protobuf//:protobuf_java",
+        artifact("com.google.code.findbugs:jsr305", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.guava:guava", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.protobuf:protobuf-java", repository_name = "contrib_rules_jvm_deps"),
     ],
 )
 
diff --git a/core/BUILD.bazel b/core/BUILD.bazel
index 1a743ff..13b51cc 100644
--- a/core/BUILD.bazel
+++ b/core/BUILD.bazel
@@ -25,13 +25,13 @@ java_library(
     deps = [
         "//api",
         "//context",
-        artifact("com.google.code.gson:gson"),
-        artifact("com.google.android:annotations"),
-        artifact("com.google.code.findbugs:jsr305"),
-        artifact("com.google.errorprone:error_prone_annotations"),
-        artifact("com.google.guava:guava"),
-        artifact("io.perfmark:perfmark-api"),
-        artifact("org.codehaus.mojo:animal-sniffer-annotations"),
+        artifact("com.google.code.gson:gson", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.android:annotations", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.code.findbugs:jsr305", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.errorprone:error_prone_annotations", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.guava:guava", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.perfmark:perfmark-api", repository_name = "contrib_rules_jvm_deps"),
+        artifact("org.codehaus.mojo:animal-sniffer-annotations", repository_name = "contrib_rules_jvm_deps"),
     ],
 )
 
diff --git a/netty/BUILD.bazel b/netty/BUILD.bazel
index 8253d1f..7543989 100644
--- a/netty/BUILD.bazel
+++ b/netty/BUILD.bazel
@@ -13,22 +13,22 @@ java_library(
     deps = [
         "//api",
         "//core:internal",
-        artifact("com.google.code.findbugs:jsr305"),
-        artifact("com.google.errorprone:error_prone_annotations"),
-        artifact("com.google.guava:guava"),
-        artifact("io.netty:netty-buffer"),
-        artifact("io.netty:netty-codec"),
-        artifact("io.netty:netty-codec-http"),
-        artifact("io.netty:netty-codec-http2"),
-        artifact("io.netty:netty-codec-socks"),
-        artifact("io.netty:netty-common"),
-        artifact("io.netty:netty-handler"),
-        artifact("io.netty:netty-handler-proxy"),
-        artifact("io.netty:netty-resolver"),
-        artifact("io.netty:netty-transport"),
-        artifact("io.netty:netty-transport-native-unix-common"),
-        artifact("io.perfmark:perfmark-api"),
-        artifact("org.codehaus.mojo:animal-sniffer-annotations"),
+        artifact("com.google.code.findbugs:jsr305", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.errorprone:error_prone_annotations", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.guava:guava", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.netty:netty-buffer", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.netty:netty-codec", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.netty:netty-codec-http", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.netty:netty-codec-http2", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.netty:netty-codec-socks", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.netty:netty-common", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.netty:netty-handler", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.netty:netty-handler-proxy", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.netty:netty-resolver", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.netty:netty-transport", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.netty:netty-transport-native-unix-common", repository_name = "contrib_rules_jvm_deps"),
+        artifact("io.perfmark:perfmark-api", repository_name = "contrib_rules_jvm_deps"),
+        artifact("org.codehaus.mojo:animal-sniffer-annotations", repository_name = "contrib_rules_jvm_deps"),
     ],
 )
 
diff --git a/protobuf-lite/BUILD.bazel b/protobuf-lite/BUILD.bazel
index 97a5e49..b4a8a8c 100644
--- a/protobuf-lite/BUILD.bazel
+++ b/protobuf-lite/BUILD.bazel
@@ -9,12 +9,10 @@ java_library(
     visibility = ["//visibility:public"],
     deps = [
         "//api",
-        artifact("com.google.code.findbugs:jsr305"),
-        artifact("com.google.guava:guava"),
-    ] + select({
-        ":android": ["@com_google_protobuf//:protobuf_javalite"],
-        "//conditions:default": ["@com_google_protobuf//:protobuf_java"],
-    }),
+        artifact("com.google.code.findbugs:jsr305", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.guava:guava", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.protobuf:protobuf-javalite", repository_name = "contrib_rules_jvm_deps"),
+    ],
 )
 
 # This config is not fully-reliable. If it breaks, it is probably because you
diff --git a/protobuf/BUILD.bazel b/protobuf/BUILD.bazel
index a31f8b6..e32c79b 100644
--- a/protobuf/BUILD.bazel
+++ b/protobuf/BUILD.bazel
@@ -10,10 +10,10 @@ java_library(
     deps = [
         "//api",
         "//protobuf-lite",
-        "@com_google_protobuf//:protobuf_java",
-        artifact("com.google.api.grpc:proto-google-common-protos"),
-        artifact("com.google.code.findbugs:jsr305"),
-        artifact("com.google.errorprone:error_prone_annotations"),
-        artifact("com.google.guava:guava"),
+        artifact("com.google.protobuf:protobuf-java", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.api.grpc:proto-google-common-protos", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.code.findbugs:jsr305", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.errorprone:error_prone_annotations", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.guava:guava", repository_name = "contrib_rules_jvm_deps"),
     ],
 )
diff --git a/stub/BUILD.bazel b/stub/BUILD.bazel
index 00e0c9c..11d6b2c 100644
--- a/stub/BUILD.bazel
+++ b/stub/BUILD.bazel
@@ -10,10 +10,10 @@ java_library(
     deps = [
         "//api",
         "//context",
-        artifact("com.google.code.findbugs:jsr305"),
-        artifact("com.google.errorprone:error_prone_annotations"),
-        artifact("com.google.guava:guava"),
-        artifact("org.codehaus.mojo:animal-sniffer-annotations"),
+        artifact("com.google.code.findbugs:jsr305", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.errorprone:error_prone_annotations", repository_name = "contrib_rules_jvm_deps"),
+        artifact("com.google.guava:guava", repository_name = "contrib_rules_jvm_deps"),
+        artifact("org.codehaus.mojo:animal-sniffer-annotations", repository_name = "contrib_rules_jvm_deps"),
     ],
 )
 
@@ -23,5 +23,5 @@ java_library(
     name = "javax_annotation",
     neverlink = 1,  # @Generated is source-retention
     visibility = ["//visibility:public"],
-    exports = [artifact("org.apache.tomcat:annotations-api")],
+    exports = [artifact("org.apache.tomcat:annotations-api", repository_name = "contrib_rules_jvm_deps")],
 )
